### PR TITLE
Create ExpandHandleSpan

### LINKED ISSUE #1421: Reuse the SignedTxnAccessor from expandSignatures in handleTransaction
**Summary**
We build a `SignedTxnAccessor` from the platform transaction in `expandSignatures`; this object could be reused in `handleTransaction` once the source `Transaction` reaches consensus.

### PR DESCRIPTION
**Pull request by tinker-michaelj**
**Related issue(s)**:
- Closes #1421

**Summary of the change**:
- Introduce the `ExpandHandleSpan` class to track context for a `SwirldTransaction` in the "span" between its use in the `SwirldState#expandSignatures` and `SwirldState#handleTransaction` callbacks.
    * At first the span only includes the `PlatformTxnAccessor` parsed from the `SwirldTransaction`.
    * This avoid re-parsing in `handleTransaction`.
    * But much more sophisticated performance optimizations can be done by adding more information to the span.

**External impacts**:
None.

**Applicable documentation**
- [x] Javadoc

### CODE REVIEW

**Review by qnswirlds — APPROVED** [review]
LGTM

**Review by anighanta — APPROVED** [review]

**Review by abhishek-hedera — APPROVED** [review]
LGTM

File: hedera-node/src/main/java/com/hedera/services/context/ServicesContext.java [hunk 1]
```diff
@@ -1640,6 +1643,13 @@ public FileUpdateInterceptor feeSchedulesManager() {
 		return feeSchedulesManager;
 	}
 
+	public ExpandHandleSpan expandHandleSpan() {
+		if (expandHandleSpan == null) {
+			expandHandleSpan = new ExpandHandleSpan(10, TimeUnit.SECONDS);
```
Comment (ljianghedera) [inline hedera-node/src/main/java/com/hedera/services/context/ServicesContext.java:1648]: Is 10 seconds too short?

File: hedera-node/src/main/java/com/hedera/services/txns/ExpandHandleSpan.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+	public PlatformTxnAccessor track(SwirldTransaction transaction) throws InvalidProtocolBufferException {
+		final var accessor = new PlatformTxnAccessor(transaction);
+		accessorCache.put(transaction, accessor);
+		return accessor;
+	}
+
+	public PlatformTxnAccessor accessorFor(SwirldTransaction transaction) throws InvalidProtocolBufferException {
+		final var cachedAccessor = accessorCache.getIfPresent(transaction);
+		return cachedAccessor != null ? cachedAccessor : new PlatformTxnAccessor(transaction);
```
Comment (ljianghedera) [inline hedera-node/src/main/java/com/hedera/services/txns/ExpandHandleSpan.java:71]: Why isn't the newly created `PlatformTxnAccessor` put into the cache?

Comment (tinker-michaelj) [reply hedera-node/src/main/java/com/hedera/services/txns/ExpandHandleSpan.java:71]: If we don't have the `PlatformTxnAccessor` cached by `handleTransaction`, then it's too late to be of any good. 😝

Comment (ljianghedera) [reply hedera-node/src/main/java/com/hedera/services/txns/ExpandHandleSpan.java:71]: Ahh, make sense. I thought there are several occasions the accessor is used.

**Pull request merged**